src/core/memory.cc

```diff
@@ -574,7 +577,8 @@ struct cpu_pages {
     std::vector<reclaimer*> reclaimers;
     static constexpr unsigned nr_span_lists = 32;
     page_list free_spans[nr_span_lists];  // contains aligned spans with span_size == 2^idx
-    small_pool_array small_pools;
+    small_pool_array<false> small_pools;
```

Comment:
Might do some more thorough testing but my napkin math agreed with yours. Hence, I agree it shouldn't be much of an issue.

Also because of the sampling, for example assuming 1MB sampling and actually sampling one of these allocations it's very likely you will actually sample more allocations of the same and actually use some of that wastage.